### ISSUE TITLE
Scale shows wrong numbers when zooming out.

### ISSUE THREAD

**Issue by zerda-ocm** [maplibre/maplibre-gl-js#4804]
**maplibre-gl-js version**: 5.0.0-pre.1

**browser**: chrome, edge

### Steps to Trigger Behavior

 1. add a scale to the map
```
var scale = new maplibregl.ScaleControl({
  maxWidth: 80,
  unit: "imperial",
});
map.addControl(scale);
```
 2a. move or zoom slightly and observe the scale
 2b. zoom out a lot and observe the scale

### Link to Demonstration

https://jsfiddle.net/zLhtufqk/

### Expected Behavior

Scale shows distance according to scale of the globe

### Actual Behavior

The scale shows for example "300mi", after a move or zoom event (even a minor one), the scale jumps to for example "1000mi". Zooming further out, the scale shows smaller and smaller numbers until the whole earth is only 1 mile wide, Zooming in seem to work fine (presumably, because the projections gets closer to mercator.)

**Comment by HarelM** [comment]
Thanks for taking the time to report this bug.
While I see the bug you have described, can you please make the stops to reproduce a bit defined for future readers?

**Comment by russss** [comment]
I just spotted this as well - you can see this behaviour on [openinframap.org](https://openinframap.org) (which is using an unmodified ScaleControl), so I think it should be reproducible by simply adding the ScaleControl to any map using the globe projection.

The scale appears to be (at least roughly) correct at zoom levels above around 3.4, but when zooming out further, the distance shown decreases rather than increases.

(To be honest it's not hugely useful at those zoom levels, so I am tempted to add a minZoom option to the ScaleControl...)

**Comment by HarelM** [comment]
Looks like it is using unproject:
https://github.com/maplibre/maplibre-gl-js/blob/c4b083fff7232630408c81b00757e55147138cf0/src/ui/control/scale_control.ts#L105
It might be that the scale location is outside the globe and thus getting unwanted values, which wasn't possible before, as the entire screen was filled with the map in 2D.

**Comment by pcardinal** [comment]
Hi,
I want to make some changes to the code to correct this issue but it is my first PR for this project. How do I link my PR to this issue ?
Thanks

**Comment by HarelM** [comment]
Usually with "Fixes #-issue" in the first PR comment.

**Comment by pcardinal** [comment]
"Fixes #4804"  ? is this ok ?

**Comment by HarelM** [comment]
Yup. There's a PR template, make sure you fill as much as you can, if you don't know simply ask.
THANKS!
